Propose VS Code Getting Started

kendrahavens opened the pull request:
# VS Code Getting Started Page
## Summary
1. Important to stay as OS independent as possible here.
2. This is modified from VS Code's docs [here ](https://code.visualstudio.com/docs/runtimes/dotnet) with a few adjustments and added pictures.
3. Linked to 3 video tutorials at bottom for respective OS.
## Fixes #1141
## Suggested Reviewers

@mairaw 

Getting started page for .NET Core, C# extension, and VS Code.

kendrahavens committed:
Propose VS Code Getting Started

Getting started page for .NET Core, C# extension, and VS Code.

dnfclas commented:
Hi __@kendrahavens__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

dnfclas commented:
__@kendrahavens__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C17%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610172229004739-1158%5Cworkflow_report.html).

kendrahavens committed:
Deleted "Next Steps" Section

Let's assume the surrounding documentation on the .NET Docs is where users will want to browse next.

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C17%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610172253477888-1158%5Cworkflow_report.html).

svick commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -15,14 +15,43 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
 
 # 🔧 Getting started with Visual Studio Code
```
The wrench here and in toc.md should be removed.

kendrahavens replied:
Gotcha.

svick commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -15,14 +15,43 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+## Getting Started
+
+1. Install [.NET Core](https://microsoft.com/net/core).
+2. Install the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) from the VS Code Marketplace.
```
Since this is not part of VS Code docs anymore, should the list include installing VS Code itself?

kendrahavens replied:
Good Call.

kendrahavens committed:
Deleted Wrench, Added VS Code Install Step

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: SucceededWithWarning. **Message**: Warning occurred: internal error: the updated_at is missing..  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C18%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610182118147767-1158%5Cworkflow_report.html).

rpetrusha commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,46 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
->
+# Getting started with Visual Studio Code
+
+.NET Core gives you a blazing fast and modular platform for creating server applications that run on Windows, Linux and macOS. Use Visual Studio Code with the C# extension to get a powerful editing experience with full support for C# IntelliSense (smart code completion) and debugging.
```
The first sentence doesn't really belong (and "blazing fast" is very marketing oriented). Most people coming here will have at least a sense of what .NET Core is and want to figure out how it works. I'd start with the second sentence, and also note the platforms on which VS Code is available -- it's not Windows-only.

kendrahavens replied:
What is in here that indicates it is Windows only?

kendrahavens committed:
Deleted "blazing" and "fun"

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610190023268763-1158%5Cworkflow_report.html).

cartermp commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,46 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+    * Go to the File Explorer Tab on the left.
+    * Open the folder you want you C# project to be in.
+
+  ![VSCodeOpenFolder](https://github.com/dotnet/core-docs/blob/master/docs/images/VSCodeOpenFolder.PNG)
```
A few things about the images:

* Since you're including media, you should create a folder in the `getting-started` directory called `media`, and place the images there.
* Reference the images as such: `![](/media/VSCodeOpenFolder.png)`

For more info about this (and more), our [Contribution Guidelines](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md#process-for-contributing) have details.

cartermp commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,46 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+    * Type `dotnet new`
+    * This creates a `Program.cs` file in your folder with a simple "Hello World" program already written.
+
+  ![dotnetNew](https://github.com/dotnet/core-docs/blob/master/docs/images/dotnetNew.PNG)
```
Same feedback as above.

cartermp commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,46 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+    * You'll see a new `project.lock.json` file in your project folder.
+    * This file contains information about your project's dependencies to make subsequent restores quicker.
+
+  ![Image dotnet restore](https://github.com/dotnet/core-docs/blob/master/docs/images/dotnetRestore.PNG)
```
Same feedback as above.

cartermp commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,46 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+4. Run the "Hello World" program by typing `dotnet run`
+
+  ![dotnetRun](https://github.com/dotnet/core-docs/blob/master/docs/images/dotnetRun.PNG)
```
Same feedback as above.

cartermp reviewed (changes requested):
Great content!  This is good to have in our docs.  Please address the issues with images - they currently won't render on staging, and don't meet our contribution guidelines.  Once that's addressed, I think this is ready to merge.

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610191843202140-1158%5Cworkflow_report.html).

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610191847431355-1158%5Cworkflow_report.html).

cartermp commented:
LGTM!  :shipit: 🚢 :it:

mairaw commented:
I think this should be under .NET Core and not the C# Guide

mairaw commented:
Ok, I just noticed we had two issues for similar things: #1141 and #944. But I still think the general getting started should be under the .NET Core guide. @BillWagner any thoughts? I don't know how you were envisioning this for your topic since you mention you'd show how to install the extension, link to other extensions, etc.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+.NET Core gives you a fast and modular platform for creating server applications that run on Windows, Linux and macOS. Use Visual Studio Code with the C# extension to get a powerful editing experience with full support for C# IntelliSense (smart code completion) and debugging.
+
+## Getting Started
```
I'd call this Prerequisites as our other topics do

kendrahavens replied:
The problem is this is a VERY specific scenario. .NET CLI tools are the main getting-started scenario for .NET. cross-platform. But this guide is only for .NET, C# in **VS Code**.

The cross-plat tutorial for .NET is already [here](https://docs.microsoft.com/en-us/dotnet/articles/core/tutorials/using-with-xplat-cli)

I don't mind if we add it under .NET Core Guide instead as a "Get started with Visual Studio Code" topic. If we did not include it in the C# guide would we also delete the "Getting Started with Visual Studio" ?

I think a lot of users who end up here are going to be trying to write C# on their mac or linux. It makes sense to have it under the C# Getting Started section in that scenario.

kendrahavens replied:
I'll change prerequisites.

mairaw replied:
My understanding from Bill's comments in #944 issue which was the one associated with this topic is that it would show the step by step to install the C# extension, provide guidance on some other extensions, etc. something more generic than your first console app in .NET Core.

kendrahavens replied:
I wasn't aware of other extensions we recommend to use with the C# one.

I'll write out instructions for installing the C# extension, but I think the VS Code install instructions are longer and very specific to the operating system and we should let the VS Code install page guide users through that. I think linking to the VS Code install page is adequate here.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
 
```
change author to your GH alias and ms.date to today's date. It'd be good to review the description too but we'll review that later as well.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+## Hello World
+
+If you'd like to get started with a simple "Hello World" program on .NET Core, follow the steps below:
```
"Hello World" program -> "Hello World" C# program?

don't use below following a noun (MSFT style guide):
follow the steps below -> use the following steps (or something like that)

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
->
+# Getting started with Visual Studio Code
+
+.NET Core gives you a fast and modular platform for creating server applications that run on Windows, Linux and macOS. Use Visual Studio Code with the C# extension to get a powerful editing experience with full support for C# IntelliSense (smart code completion) and debugging.
```
I'd probably call out that in this tutorial we'll be showing how to create your first console app in C# since you do have other options.

kendrahavens replied:
I'll add C#, but not the console app. That is assumed with the Hello World heading.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+1. Open a project:
+
+    * Open VS Code.
+    * Go to the File Explorer Tab on the left.
```
I still feel that it would be better to call out the steps via menu UI. If I open VS Code and click on the Explorer icon, I might have a project already opened and so I don't always get the Open Folder button. If you do **File**, **Open Folder**, the result will be predictable. Unless you don't have this option in all OSes.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+  ![VSCodeOpenFolder](media/with-visual-studio-code/VSCodeOpenFolder.PNG)
+
+2. Initialize a C# project:
+    * Open the command prompt (or terminal). You can open the terminal from VS Code by typing <kbd>CTRL</kbd>+<kbd>`</kbd> (Back Tick)
```
In Vs Code, it's called Integrated Terminal no?

Back Tick -> backtick 

also OPS doesn't do anything special with the <kbd> tag, so I'm wondering if we'd want to style this a bit different (see the link to the staged content)

and backtick when used as text should be escaped, otherwise the rest of the text is not formatted properly: 
```markdown
\`
``` 

Last but not least, if we want to stay cross-platform, we need to either give the key combinations for all OSes or perhaps use the menu UI again to open the terminal.

kendrahavens replied:
The key combinations are the same from all OSes.

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+2. Initialize a C# project:
+    * Open the command prompt (or terminal). You can open the terminal from VS Code by typing <kbd>CTRL</kbd>+<kbd>`</kbd> (Back Tick)
+    * Type `dotnet new`
+    * This creates a `Program.cs` file in your folder with a simple "Hello World" program already written.
```
Wondering if it would be good to call out here the reference topic for the dotnet new command so folks can learn how they can customize the command to create F# projects or other projects types. It can be simple as 
"For more information, see the [dotnet-new](relative link) topic."

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610192121018799-1158%5Cworkflow_report.html).

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+3. Resolve the build assets by typing `dotnet restore`
+
+    * Running `restore` pulls down the required packages declared in the `project.json` file.
```
`restore` -> `dotnet restore`

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+    * Running `restore` pulls down the required packages declared in the `project.json` file.
+    * You'll see a new `project.lock.json` file in your project folder.
+    * This file contains information about your project's dependencies to make subsequent restores quicker.
```
link to dotnet restore topic here?

mairaw commented on docs/csharp/getting-started/with-visual-studio-code.md:
```diff
@@ -13,16 +13,45 @@ ms.assetid: 76c23597-4cf9-467e-8a47-0c3703ce37e7
+
+  ![dotnetRun](media/with-visual-studio-code/dotnetRun.PNG)
+
+You can also watch a short video tutorial for further setup help on [Windows](https://channel9.msdn.com/Blogs/dotnet/Get-started-with-VS-Code-using-CSharp-and-NET-Core), [macOS](https://channel9.msdn.com/Blogs/dotnet/Get-started-with-VS-Code-using-CSharp-and-NET-Core-on-MacOS), or [Linux](https://channel9.msdn.com/Blogs/dotnet/Get-started-with-VS-Code-Csharp-dotnet-Core-Ubuntu).
```
Should we add a See Also header for this? I think it'd add visibility to these links.

kendrahavens replied:
Maira, with all these small changes I'm confused why you didn't just make a commit. Did I set up my branch wrong? Are you able to make commits?

BillWagner commented:
@mairaw I'd like to see this include instructions on using the debugger (at a getting started level).

I do think this content could fit under both the .NET Core Guide, and the C# Guide. Ideally, we'll have the virtual TOC feature we've talked about. In the meantime, I'd prefer C#, as we have more 'getting started' content in Core, and less around C#. But, I'm fine either way.

mairaw commented:
Sounds good @BillWagner!

kendrahavens commented:
A debugging section would be good. It should have pictures too.

qinezh commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C10%5C21%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201610210111461136-1158%5Cworkflow_report.html).

BillWagner commented:
@kendrahavens Can you rebase this and fix the merge conflicts? Then, it should be ready to go.

kendrahavens committed:
Add images

kendrahavens committed:
Fixed Pic links

kendrahavens committed:
Fixed old commits

saldana commented:
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158)  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio?branch=pr-en-us-1158](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio?branch=pr-en-us-1158)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C3%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611032116104212-1158%5Cworkflow_report.html).

kendrahavens committed:
Fixed merge conflicts

cartermp commented on docs/csharp/getting-started/with-visual-studio.md:
```diff
@@ -1,5 +1,6 @@
 ---
 <<<<<<< HEAD
+<<<<<<< HEAD
```
Looks like merge conflict indicators were still left in.

kendrahavens replied:
That doc gave me trouble for some reason even though the names are totally different. Should be fixed now.

saldana commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/getting-started/with-visual-studio-code.md](https://github.com/dotnet/docs/blob/kendrahavens-patch-1/docs/csharp/getting-started/with-visual-studio-code.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158) | [Details](#user-content-docs/csharp/getting-started/with-visual-studio-code.md)
[docs/csharp/getting-started/with-visual-studio.md](https://github.com/dotnet/docs/blob/kendrahavens-patch-1/docs/csharp/getting-started/with-visual-studio.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio?branch=pr-en-us-1158) | [Details](#user-content-docs/csharp/getting-started/with-visual-studio.md)



For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C14%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611142021201973-1158%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

kendrahavens committed:
VS page conflict fix

saldana commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/csharp/getting-started/with-visual-studio-code.md](https://github.com/dotnet/docs/blob/kendrahavens-patch-1/docs/csharp/getting-started/with-visual-studio-code.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio-code?branch=pr-en-us-1158) | [Details](#user-content-docs/csharp/getting-started/with-visual-studio-code.md)
[docs/csharp/getting-started/with-visual-studio.md](https://github.com/dotnet/docs/blob/kendrahavens-patch-1/docs/csharp/getting-started/with-visual-studio.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/articles/csharp/getting-started/with-visual-studio?branch=pr-en-us-1158) | [Details](#user-content-docs/csharp/getting-started/with-visual-studio.md)



For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C14%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201611142218541906-1158%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

cartermp commented:
LGTM.  :shipit:!

mairaw commented:
LGTM!
